Add reactive MS SQL client extension

tsegismont opened the pull request:
- runtime/deployment modules
- integration tests (including native mode)
- datasources docs

tsegismont committed:
MS SQL extension

- runtime/deployment modules
- integration tests (including native mode)
- datasources docs

tsegismont commented:
cc @Sanne @DavideD

tsegismont committed:
Updates after running update-extension-dependencies.sh

cescoffier commented on extensions/reactive-mssql-client/deployment/pom.xml:
```diff
@@ -0,0 +1,200 @@
+
+    <artifactId>quarkus-reactive-mssql-client-deployment</artifactId>
+
+    <name>Quarkus - Reactive MS SQL Client - Deployment</name>
```
Same question about the name.
It looks from their page that the name is just "SQL Server" but SQL Server Client is weird... SQL Client is not exact. I understand how you ended up with MS SQL Client. I would still expand MS.

Sanne replied:
+1 to expand here and in the description for sake o clarity.

In other areas the short hand is fine, as it's commonly referred to as MS SQL by its users (which we are addressing).

tsegismont replied:
Done

cescoffier commented on extensions/reactive-mssql-client/deployment/src/test/java/io/quarkus/reactive/mssql/client/DevModeResource.java:
```diff
@@ -0,0 +1,42 @@
+            if (ar.succeeded()) {
+                future.complete(Response.serverError().entity("Expected SQL query to fail").build());
+            } else if (!expectedExceptionClass.isAssignableFrom(ar.cause().getClass())) {
+                ar.cause().printStackTrace();
```
should be removed.

tsegismont replied:
Done

gsmet commented on docs/src/main/asciidoc/datasource.adoc:
```diff
@@ -662,6 +662,10 @@ include::{generated-dir}/config/quarkus-reactive-db2-client.adoc[opts=optional,
 
 include::{generated-dir}/config/quarkus-reactive-mysql-client.adoc[opts=optional, leveloffset=+1]
 
+=== Reactive MS SQL Specific Configuration
```
It's Microsoft SQL Server.

tsegismont replied:
Done

Sanne commented:
@DavideD could you follow up with integration tests using MS SQL + Hibernate Reactive ?
Which I assume will need an HR release ?

DavideD commented:
Do you mean integration tests for Quarkus? Sure, [we are having somes trouble with the MS SQL Client on Hibernate Reactive](https://github.com/hibernate/hibernate-reactive/pull/865#issuecomment-872419889) at the moment, but as soon as it has been fixed, I will add them

tsegismont committed:
Expand Microsoft SQL Server name

tsegismont committed:
Remove redundant stacktrace from the test console

tsegismont committed:
MSSQL Reactive Client SSL not supported yet

tsegismont committed:
TDS Packet size configurable with Vert.x 4.1.1

tsegismont commented:
Thank you all for the review. I addressed the concerns and rebased on `main`. This allowed to add support for the TDS packet size option that comes with Vert.x 4.1.1.

Sanne commented:
Looks like the failures are related:
```
2021-07-02T18:09:30.9633222Z [ERROR] Failed to execute goal io.fabric8:docker-maven-plugin:0.31.0:start (docker-start) on project quarkus-integration-test-reactive-mssql-client: I/O Error: [mcr.microsoft.com/mssql/server:2019-latest] "quarkus-test-mssql": Timeout after 10045 ms while waiting on tcp port '[/172.17.0.2:1433]' -> [Help 1]
```

Sanne commented:
it's a DB bootstrap timeout:
```
2021-07-02T17:29:38.5268124Z [ERROR] DOCKER> [mcr.microsoft.com/mssql/server:2019-latest] "quarkus-test-mssql": Timeout after 10045 ms while waiting on tcp port '[/172.17.0.2:1433]'
```

Sanne commented:
@tsegismont could you copy the infrastructure from the pom.xml in `integration-tests/jpa-mssql` ? It seems your pom.xml is similar but doesn't have crucial elements like the timeout settings.

cescoffier reviewed (approved):
It looks good to me. However, validation by Sanne would be great (as I don't know much about databases anyway).

tsegismont commented:
>@tsegismont could you copy the infrastructure from the pom.xml in integration-tests/jpa-mssql ? It seems your pom.xml is similar but doesn't have crucial elements like the timeout settings.

@Sanne good idea, I will do it for both the integration tests and the deployment module tests.

tsegismont committed:
Update MSSQL container setup to avoid timeouts

Took example from JPA-MSSQL integration tests module.

Sanne commented on extensions/reactive-mssql-client/deployment/pom.xml:
```diff
@@ -143,13 +140,9 @@
-                                            </tcp>
+                                            <log>SQL Server is now ready for client connections</log>
+                                            <!-- Unfortunately booting this is slow, needs to set a generous timeout: -->
+                                            <time>40000</time>
```
:+1:  thanks! Let's see if CI managed to run it now.

tsegismont replied:
There is one failure but it looks unrelated

Sanne commented:
awesome, let's merge this. Many thanks!
